### ISSUE TITLE
Logger inside event handler in circuit breaker printing multiple times

### ISSUE THREAD

**Issue by gauravarya604** [resilience4j/resilience4j#1102]
Resilience4j version:1.5

Java version:8

Issue: I have initiated 150 calls, where function wrapped in circuit breaker always throws an error. Configs are like that circuit will always be closed and each call will return an error.

Here is the code
```
 circuitBreaker.getEventPublisher()
                .onError(event -> {
                    logger.error("error encountered in circuit breaker :{}",event.getThrowable().getMessage());
                }));
    }
```
**Here is the log that I am receiving from the inside circuit breaker.**
`I have this line **" error encountered in circuit breaker"** is being printed 6500 times  in logs`

If I call this one time, the above line is printed 4,5 times in logs.

**Comment by mehwhatever** [comment]
Are you sure you have done invoked `circuitBreaker.getEventPublishe.onError` just once ?

**Comment by gauravarya604** [comment]
Yes i am triggering it only once. And moreover each time i reinitiate circuit breaker and again trigger 150 calls, the count of logs printed id always different.

**Comment by Romeh** [comment]
@gauravarya604  do u have code example or unit test code sample i can check ?

**Comment by bhukailas** [comment]
in my production it printed 2466 times

**Comment by sgomatam18** [comment]
I am logging state transition and I am also getting multiple logs for each transition. circuitBreaker.getEventPublisher().onStateTransition(event -> log.info(event.toString())

**Comment by amitranjan3** [comment]
@bhukailas @sgomatam18 have you find the solution ? i am also facing the same issue
Thanks & Regards

**Comment by RobWin** [comment]
When and where are you adding this code snippet?
Did you verify that your code snippet is not invoked multiple times?

```
circuitBreaker.getEventPublisher()
                .onError(event -> {
                    logger.error("error encountered in circuit breaker :{}",event.getThrowable().getMessage());
                }));
    }
```

**Comment by amitranjan3** [comment]
Hi @RobWin i am adding this code in configuration and facing same  multiple log issue
<img width="1054" alt="Screenshot 2022-03-18 at 11 55 32 PM" src="https://user-images.githubusercontent.com/99726415/159061727-8de4cd8f-2741-4a40-b4a6-f1c527a68e93.png">

**Comment by sgomatam18** [comment]
It is not resolved yet.
    On Thursday, March 17, 2022, 07:19:34 PM EDT, amitranjan3 ***@***.***> wrote:  
 
 


@bhukailas @sgomatam18 have you find the solution ? i am also facing the same issue
Thanks & Regards

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by Mahesh-PP** [comment]
I am also facing the same issue 

   CircuitBreaker circuitBreaker = registry
                .circuitBreaker(params.getCarrierCode()+params.getRequestType(), "AAgetDetailsV3");

        circuitBreaker.getEventPublisher().onStateTransition(e -> System.out.println("onStateTransition : "+ e.toString()));

**Comment by himanshujain1985** [comment]
We are also facing this same issue. Did we ever find any workaround to this problem?

**Comment by hi-im-bob** [comment]
Facing the same problem using spring-cloud-starter-circuitbreaker-resilience4j for state transition event consumer.

`Customizer#once` solved the issue. See https://github.com/spring-cloud/spring-cloud-circuitbreaker/issues/67#issuecomment-623334998

**Comment by bhukailas** [comment]
@sgomatam18 you need need to ensure following code is called only once for a given circuit breaker object and you need to do this as onError, onIgnoredError, onReset etc does register more than one lamda if called more than once and this was causing duplicate errors in my case and this even causes CPU throttling too.

circuitBreaker.getEventPublisher()
    .onSuccess(event -> logger.info(...))
    .onError(event -> logger.info(...))
    .onIgnoredError(event -> logger.info(...))
    .onReset(event -> logger.info(...))
    .onStateTransition(event -> logger.info(...));
